src/system/connectivity/wifi.cpp

```diff
@@ -10,35 +12,27 @@ static String templateID;
 static String deviceName;
 static String token;
 
-static void setPreSaveConfigCallback()
+static void saveConfigCallback()
 {
-    if(wm.server->hasArg("template_id"))
+    logger << LOG_INFO << "Callback called" << EndLine; //REMOVE
+    if((wm.server->hasArg("template_id")) || (wm.server->hasArg("device_name")) || (wm.server->hasArg("auth_token")))
```

Comment:
This conditional is wrong, should check that all the parameters exists, because you're salving  the three parameters.

Reply:
good catch